### PR TITLE
fix two edge cases in equivalence scorer

### PR DESCRIPTION
**Pull request by AdamDM**
Edge case 1 - title starts with more than one ':' throwing OOB
Edge case 2 - title starts with a number followed by a colon and matches a regex throwing OOB

Due to these changes, titles with a leading ':' may be less likely to get a partial match, but these should be rare anyway. These changes are reflected in the test alterations

### CODE REVIEW

File: src/main/java/org/atlasapi/equiv/scorers/TitleMatchingItemScorer.java [hunk 1]
```diff
@@ -160,33 +160,39 @@ private Score compareTitles(final String subjectTitle, final String suggestionTi
...
 
         return scoreOnMismatch;
     }
 
+    private String removeLeadingColons(String title) {
+        Pattern colonsAtStart = Pattern.compile(":+(.*)");
+
+        Matcher titleMatcher = colonsAtStart.matcher(title);
+
+        return titleMatcher.matches() ? titleMatcher.group(1) : title;
```
Comment (jamiesprax) [inline src/main/java/org/atlasapi/equiv/scorers/TitleMatchingItemScorer.java]: This looks like what `String.replace` does. Assuming the string isn't null/empty:
`if (title.charAt(0) == ':') {
    return title.replaceFirst(":", "");
}`

File: src/test/java/org/atlasapi/equiv/scorers/TitleMatchingItemScorerTest.java [hunk 2]
```diff
@@ -199,9 +199,20 @@ public void testPartialMatchAfterSemicolon() {
...
+    public void testForMultipleColonsToStart() {
+        DefaultDescription desc = new DefaultDescription();
+        score(0, scorer.score(itemWithTitle("Storage Hunters"), of(itemWithTitle("::::Storage Hunters: UK")), desc));
     }
 
+    @Test
+    public void testForPrefixRemovalBug() {
+        DefaultDescription desc = new DefaultDescription();
+        score(0, scorer.score(itemWithTitle("Storage: Bunters"), of(itemWithTitle(" 5 : Storage Hunters")), desc));
+    }
```
Comment (jamiesprax) [inline src/test/java/org/atlasapi/equiv/scorers/TitleMatchingItemScorerTest.java]: What would happen if `title = "::::"` ?

File: src/main/java/org/atlasapi/equiv/scorers/TitleMatchingItemScorer.java (hunk 1 above)
Comment (AdamDM) [inline src/main/java/org/atlasapi/equiv/scorers/TitleMatchingItemScorer.java]: It's meant to prevent the edge case of something starting with multiple ':'s from throwing exceptions later in the code

File: src/test/java/org/atlasapi/equiv/scorers/TitleMatchingItemScorerTest.java [hunk 3]
```diff
@@ -199,9 +199,20 @@ public void testPartialMatchAfterSemicolon() {
     @Test
     public void testForOutOfBoundsException() {
         DefaultDescription desc = new DefaultDescription();
-        score(1, scorer.score(itemWithTitle("Storage Hunters"), of(itemWithTitle(":Storage Hunters: UK")), desc));
+        score(0, scorer.score(itemWithTitle("Storage Hunters"), of(itemWithTitle(":Storage Hunters: UK")), desc));
```
Comment (Dalamar42) [inline src/test/java/org/atlasapi/equiv/scorers/TitleMatchingItemScorerTest.java]: All these tests are returning a score of 0. You need some tests that return a good score for these cases as well
